relay-profiling/src/lib.rs

```diff
@@ -126,26 +128,48 @@ struct MinimalProfile {
 }
 
 fn minimal_profile_from_json(payload: &[u8]) -> Result<MinimalProfile, ProfileError> {
-    serde_json::from_slice(payload).map_err(ProfileError::InvalidJson)
+    let d = &mut Deserializer::from_slice(payload);
+    serde_path_to_error::deserialize(d).map_err(ProfileError::InvalidJson)
 }
 
 pub fn parse_metadata(payload: &[u8]) -> Result<(), ProfileError> {
     let profile = match minimal_profile_from_json(payload) {
         Ok(profile) => profile,
-        Err(err) => return Err(err),
+        Err(err) => {
+            relay_log::warn!(error = &err as &dyn Error, "invalid minimal profile");
```

Comment:
nit: Why did you move the warning to `relay-profiling`? Is it to know the difference between an error in minimum profile parsing and an error in sample parsing? For that, we could also use separate `ProfileError::InvalidMinimalJson` and `ProfileError::InvalidSampleJson`. That way, we could remove the `relay-log` dependency again.

Reply:
The reason I did this is to be able to expose the platform and possible version of the profiling schema in the error in the actual parsing of the profile. I don't necessarily care about differentiating minimal vs full parsing error, it's just something I have to do if I want to show the platform.